### DISCUSSION TITLE
"On This Day" filter

### DISCUSSION THREAD

**Discussion by mystery-house** [usememos/memos#3913] [Ideas]
I'm trying out Memos as a replacement for the [DayOne](https://dayoneapp.com/) personal journaling app, and I'm really enjoying how clean and basic it is both in terms of UI and deployment.

A DayOne feature that I used a lot was the "On This Day" filter, which would display a list of entries filtered by the current month and date; so for example if today was 2024-09-01, clicking "On This Day" would show entries from 2024-09-01, 2023-09-01, 2023-09-01, et cetera.

I'm still getting acquainted with Memo's source code and APIs so I could be wrong here, but my sense is that implementing a feature like this efficiently would require adding some additional options to [MemoFilter](https://github.com/usememos/memos/blob/42bc769d5d8f80d5a9f9a64c71d9792b1f025653/server/router/api/v1/memo_service.go#L975), something like `display_time_month` and `display_time_day` (and might as well add `display_time_year` too?) That way an 'On This Day' button could be easily added to the front-end using the month and day filters. (Alternately, I suppose `display_time_before` and `display_time_after` could be extended to accept multiple values, and the front-end would have to calculate and pass those timestamps for each day going back _n_ years.

If a PR for this feature would be welcome I'd be willing to have a go at it.

**Comment by Ltaimao** [comment]
I need this feature too, please let me know if it gets implemented. The deployment version I'm using right now is this one: https://github.com/jkjoy/memos-on-cloudflare

**Comment by boojack** [comment]
This works with the CEL filter now — as of latest `main`, timestamp accessors also work on `now`, so it can be saved as a shortcut that re-resolves at query time:

**On this day:**

```cel
created_ts.getMonth() == now.getMonth() && created_ts.getDate() == now.getDate()
```

**Previous years only (DayOne-style):**

```cel
created_ts.getMonth() == now.getMonth() && created_ts.getDate() == now.getDate() && created_ts.getFullYear() < now.getFullYear()
```

Note: `getMonth()` is 0-based and date parts are extracted in UTC.
